src/layers/invertible_layer_glow.jl

```diff
@@ -79,11 +79,21 @@ end
 CouplingLayerGlow(C::Conv1x1, RB::FluxBlock; logdet=false, activation::ActivationFunction=SigmoidLayer()) = CouplingLayerGlow(C, RB, logdet, activation)
 
 # Constructor from input dimensions
-function CouplingLayerGlow(n_in::Int64, n_hidden::Int64; freeze_conv=false, k1=3, k2=1, p1=1, p2=0, s1=1, s2=1, logdet=false, activation::ActivationFunction=SigmoidLayer(), ndims=2)
+function CouplingLayerGlow(n_in::Int64, n_hidden::Int64; nx=nothing, dense=false, freeze_conv=false, k1=3, k2=1, p1=1, p2=0, s1=1, s2=1, logdet=false, activation::ActivationFunction=SigmoidLayer(), ndims=2)
 
     # 1x1 Convolution and residual block for invertible layer
     C = Conv1x1(n_in; freeze=freeze_conv)
-    RB = ResidualBlock(Int(n_in/2), n_hidden; k1=k1, k2=k2, p1=p1, p2=p2, s1=s1, s2=s2, fan=true, ndims=ndims)
+   
+    split_num = Int(round(n_in/2))
+    in_chan   = n_in-split_num
+    out_chan  = 2*split_num
```

Comment:
Don't think that's correct for `n_in=1`, one of the two in_chan, out_chan gonna be zero

Reply:
You are right, and I will add a check to have a graceful error. But Im thinking that because we are currently splitting on channel in the couplying layer so always will need at least two channels. this is why the split scales needs to be turned on when the original input only has one channel, that way you artificially create more channels.

Reply:
Oh right yes there is the coupling layer that splits the channel. I think rather than an error I'd just auto turn-on split-scale for single channel since that's what the user would want, the number of channel